frontend/src/pages/CompareV2.tsx

```diff
@@ -15,9 +15,95 @@
  */
 
 import React from 'react';
+import { useQuery } from 'react-query';
+import { ApiRunDetail } from 'src/apis/run';
+import { QUERY_PARAMS } from 'src/components/Router';
 import { commonCss } from 'src/Css';
+import { Apis } from 'src/lib/Apis';
+import { URLParser } from 'src/lib/URLParser';
+import { errorToMessage } from 'src/lib/Utils';
+import {
+  getExecutionsFromContext,
+  getKfpV2RunContext,
+  getOutputLinkedArtifactsInExecution,
+  LinkedArtifact,
+} from 'src/mlmd/MlmdUtils';
+import { Execution } from 'src/third_party/mlmd';
+import { PageProps } from './Page';
+
+interface ExecutionArtifacts {
+  execution: Execution;
+  linkedArtifacts: LinkedArtifact[];
+}
+
+interface RunArtifacts {
+  run: ApiRunDetail;
+  executionArtifacts: ExecutionArtifacts[];
+}
+
+function CompareV2(props: PageProps) {
+  const queryParamRunIds = new URLParser(props).get(QUERY_PARAMS.runlist);
+  const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
+
+  // Retrieves run details.
+  const { data: runs } = useQuery<ApiRunDetail[], Error>(
+    ['run_details', { ids: runIds }],
+    () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
+    {
+      staleTime: Infinity,
+      onError: async error => {
+        const errorMessage = await errorToMessage(error);
+        props.updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message: `Error: failed loading ${runIds.length} runs. Click Details for more information.`,
+          mode: 'error',
+        });
+      },
+      onSuccess: () => props.updateBanner({}),
+    },
+  );
+
+  // Retrieves MLMD states (executions and linked artifacts) from the MLMD store.
+  const { data: runArtifacts } = useQuery<RunArtifacts[], Error>(
+    ['run_artifacts', { runIds, runs }],
+    () => {
+      if (runs) {
+        return Promise.all(
+          runIds.map(async (r, index) => {
+            const context = await getKfpV2RunContext(r);
+            const executions = await getExecutionsFromContext(context);
+
+            const executionArtifacts = await Promise.all(
+              executions.map(async execution => {
+                const linkedArtifacts = await getOutputLinkedArtifactsInExecution(execution);
+                return {
+                  execution,
+                  linkedArtifacts,
+                } as ExecutionArtifacts;
+              }),
```

Comment:
Before deep diving to your PR, let's discuss the MLMD pulling strategy here:

Number of MLMD call will grow by the number of `Executions` in a run. If you are going to pull everything in MLMD, I suggest to make 4 MLMD calls per run: Context, Executions, Artifacts and Events. Example: https://github.com/kubeflow/pipelines/blob/d11b097a08c55543ff6d856fd58896422c250784/frontend/src/pages/RunDetailsV2.tsx#L105-L108. That means you need to make at most 40 MLMD calls -> 4 MLMD calls for 10 runs.

How to connect execution and artifact?
Event contains execution_id and artifact_id: https://github.com/google/ml-metadata/blob/master/ml_metadata/proto/metadata_store.proto#L224-L228. So when you want to find the execution for an artifact: You can find the `event` which `output` this artifact using `artifact ID` and `event type`, then you can get `execution` using `execution_id` in the `event`. 

_Even more, you can get all contexts with one MLMD call: https://github.com/google/ml-metadata/blob/master/ml_metadata/proto/metadata_store_service.proto#L676-L681. But this optimization is minimal, so it is just FYI._

Reply:
I've made the fix here: https://github.com/kubeflow/pipelines/pull/7897/commits/a2a7f7ae4bf0c66b645af4489ddb4c8fed5b6c3e. 

Note: There are a few commits after this one to resolve merge conflicts and centralize banner update logic.